app/waf/src/controllers/schema.ts

```diff
@@ -258,26 +226,28 @@ export class AdcSchema {
 
   static adcCreateRequest = buildRequestSchema(
     'ADC resource that need to be created',
-    ['type', 'host'],
-    adcCreateRequestProperties,
-    adcCreateRequestExample,
+    adcProperties.required,
```

Comment:
The aim of refactoring is removing ANY extra works of doc for a specific model. 
So I expect to remove any references/codes for a specific model, like here `adc`. 
Or else, we have to add more when 'doc' more other models, like `Application` `Endpointpolicy` ... 

What I imagine the implementation could be: 

```
  @post(prefix + '/adcs', {
    responses: {
      '200': {
        description: 'Successfully create ADC resource',
        content: {'application/json': Schema.getResponse(Adc)},
      },
      '400': {
        description: 'Invalid ADC resource',
        content: {'application/json': Schema.getBadRequest(Adc)},
      },
      '422': {
        description: 'Unprocessable ADC resource',
        content: {'application/json': Schema.getUnprocessableEntity(Adc)},
      },
    },
  })
  async create(
    @requestBody(Schema.getCreateRequest(Adc)) reqBody: Partial<Adc>,
  ): Promise<AdcResponse> {
    try {
...
```

What's more, I hope you can move the schema.ts out of controllers folder, it's not a controller. 
It's not necessary to put them together for referring. Make the code as clean as possible.

Reply:
Let me have a try to make it a little more simpler.

And I think currently there is not any other better place to store schema.ts. Since it is part of controller initializing facilities, it does make sense to place it under controller folder.

Reply:
How about putting it under `src` directly, it's actually not related with controller but document. Just like `log4ts.ts` `components.ts` `keys.ts` ... In the future if we think there are too many .ts under `src`, we can categorize them then, which is not a big job.

Reply:
Let us focus on other constructive topics.

Reply:
Just wonder that is there any special reason that you stick on keeping the file in controllers?